app/controllers/api/v1/rooms_controller.rb

```diff
@@ -5,11 +5,10 @@ class RoomsController < Api::BaseController
       respond_to :json
 
       def index
-        if params[:conference_id].blank?
-          rooms = Room.all
+        if @conference
+          @conference.venue ? (rooms = @conference.venue.rooms) : (rooms = [])
```

Comment:
please don't assign variables in conditionals. This is really hard to read! Instead do sth like:

``
rooms = @conference.venue ? ...
``

Furthermore please don't return an empty Array, instead use ``Room.none`` to stick to the ActiveRecord API.

Long answer, short: Please change the line to:

```
rooms = @conference.venue ? @conference.venue.rooms : Room.none
```